### DISCUSSION TITLE
REGCA1

### DISCUSSION THREAD

**Discussion by cqss11** [CURENT/andes#458] [Data Format]
Suspect initialization issue! Simulation may crash!

    Name     | Var. Value | Eqn. Mismatch
-------------+------------+--------------
 Qe REGCA1 1 | 0.040      | -0.040
 
Hi, hantao.
I develop a new model with REGCA1 and this error how can fix it？

**Comment by cuihantao** [comment]
Hi @cqss11 ,

There is too little information for me to tell. Do you have another generator connected to the same bus as this REGCA1? If so, check if `gammaq` is set properly, i.e., all the `gammaq` of generators connected to the same bus need to add up to 1.

**Reply by cqss11** [reply]
[15BUS_V1_Wind.xlsx](https://github.com/CURENT/andes/files/11607992/15BUS_V1_Wind.xlsx)
This is my model. And there is the initialization and out of limits issue.
Should I change the p0/q0 parameter of PQ/PV/SLACK？

**Reply by cuihantao** [reply]
I couldn't do this for you but below are the general steps to follow:

1. Use `andes run 15BUS_V1_Wind.xlsx -r tds --init &> init_log.txt` to save the variable initialization log to a file.
2. By checking the documentation of `REGCA1`, we can tell that `Qe` is initialized to `Q0`, which is `gammaq` times the `q0` of the corresponding static equation. The algebraic equation corresponding to `Qe` is `v * Iqout_y`. The mismatch complained by the program is between the values of `Qe` and `v * Iqout_y`. 
3. Inspect `init_log.txt` and trace how `v` and `Iqout_y` are assigned.

I quickly checked that `Iqout_y` has to do with a limiter. You can back calculate the correct `Iqout_y` to make `Qe` and `v * Iqout_y` match. Then back track how `Iqout_y` is calculated -- might need to read equations in the code as well.

**Reply by cuihantao** [reply]
You can also do this interactively by running `andes run 15BUS_V1_Wind.xlsx -s`, which will take you to an iPython shell. Then, do `system.TDS.init()`.

In the iPython shell, you can explore the values of the variables, e.g., by inspecting `system.REGCA1.Iqref_y`
